### PR TITLE
implemented celfpp coverage computation

### PR DESCRIPTION
**Pull request by jjboo**
Added the coverage computation (mg1, mg2) for CELF++.
Refactored IC.java

### CODE REVIEW

File: java/src/ca/ubc/model/IndependentCascade.java [hunk 1]
```diff
@@ -17,10 +17,96 @@
...
+
+    double[] ret = new double[2];
+
+    for (int i = 0; i < config.mcRuns; ++i) {
+      // reset relevant data structures
+      Arrays.fill(active, false);
+      bfsQueue.clear();
+      int countActive = 0;
+
+      // activate the candidate
```
Comment (amitgoyal12) [inline java/src/ca/ubc/model/IndependentCascade.java:66]: I thought you wanted to allow candidate == null. No?

File: java/src/ca/ubc/model/IndependentCascade.java [hunk 2]
```diff
@@ -17,10 +17,96 @@
...
+
+      // activate all seeds
+      for (int s : seeds) {
+        countActive++;
+        active[s] = true;
+        bfsQueue.add(s);
+      }
+
+      // compute MG(u | S)
+      countActive += bfs(bfsQueue, graph, active);
```
Comment (amitgoyal12) [inline java/src/ca/ubc/model/IndependentCascade.java:79]: bfs is a bad name for the method. what exactly this method is doing?

Comment (jjboo) [reply java/src/ca/ubc/model/IndependentCascade.java:79]: Breadth-first search

File: java/src/ca/ubc/model/IndependentCascade.java (hunk 1 above)
Comment (jjboo) [reply java/src/ca/ubc/model/IndependentCascade.java:66]: For CELF++ it's not needed. Because vanilla spread computation will never call this method.

**Pull request closed without merging**